Pendulum/Data/Export.swift

```diff
@@ -32,19 +64,148 @@ struct ExportedPenPal: Codable {
     let id: UUID
     let name: String
     let initials: String
-    let image: Data?
     let notes: String?
-    let archived: Bool
     let events: [ExportedEvent]
+    let archived: Bool
+    
+    init(from: PenPal, in context: NSManagedObjectContext) {
+        self.id = from.id ?? UUID()
+        self.name = from.wrappedName
+        self.initials = from.wrappedInitials
+        self.notes = from.notes
+        self.archived = from.archived
+        self.events = from.events(from: context).map { ExportedEvent(from: $0) }
+    }
+    
+    func loadImage(fromArchive archiveDirectory: URL) -> Data? {
+        do {
+            return try Data(contentsOf: archiveDirectory.appendingPathComponent("penpal-\(id.uuidString)").appendingPathExtension("png"))
+        } catch {
+            appLogger.debug("Could not load contact image \(id) from archive \(archiveDirectory)")
+        }
+        return nil
+    }
+    
 }
 
 struct ExportedStationery: Codable {
     let id: UUID
     let type: String
     let value: String
+    
+    init(from: Stationery) {
+        self.id = from.id ?? UUID()
+        self.type = from.wrappedType
+        self.value = from.wrappedValue
+    }
+    
+}
+
+struct ImportResult {
+    let stationeryCount: Int
+    let penPalCount: Int
+    let eventCount: Int
+    let photoCount: Int
 }
 
 struct Export: Codable {
     let penpals: [ExportedPenPal]
     let stationery: [ExportedStationery]
+    var context: NSManagedObjectContext? = nil
+    
+    enum CodingKeys: CodingKey {
+        case penpals, stationery
+    }
+    
+    init(from context: NSManagedObjectContext) {
+        self.penpals = PenPal.fetchAll(from: context).map { ExportedPenPal(from: $0, in: context) }
+        self.stationery = Stationery.fetch(from: context).map { ExportedStationery(from: $0) }
+        self.context = context
+    }
+    
+    func asJSON() throws -> Data {
+        let encoder = JSONEncoder()
+        return try encoder.encode(self)
+    }
+    
+    var name: String {
+        let dateFormatter = DateFormatter()
```

Comment:
Creating a DateFormatter for every single time is somewhat expensive, it's usually better to (lazily) create and reuse a single DateFormatter